src/awx_plugins/interfaces/_temporary_private_api.py

```diff
@@ -4,6 +4,52 @@
 The hope is that it will be refactored into something more standardized.
 """
 
+from collections.abc import Callable
+
+
+class Credential:
```

Comment:
I recommend putting this into a separate module `_temporary_private_credential_api.py` to prevent possible circular import loops in the future. Seeing that it's set as a type in `awx.main.models.credential` during the type checking time, makes me think that this will happen once you attempt to import this in actual awx and do the same later.